SNS.WEBSCRAPER/main.py

```diff
@@ -0,0 +1,178 @@
+        self.check_urls = check_urls
+        self.wideo_link = wideo_link
+        self.wiadomosc_link = wiadomosc_link
+        self.page_content = []
+        self.content_urls = []
+        self.final_urls = []
+        self.lenght = len(self.scraped)
+        self.particular_page_content = []
+        
+        #taking out all titles that are longer then 30 characters
+        for eliminate_invalid in self.scraped:
+            if len(eliminate_invalid.text) > 30:
+                self.page_content.append(eliminate_invalid.text)
+        
+    # Scarping data with input equal to "polsat"
+    def polsat_scrap(self):
+        websiteLinks = "Polsat News"
+        
+        #elimination of invalid scraps
+        for link in self.results.find_all('a'):
+            self.content_urls.append(link.get('href'))
+            
+        #elimiating duplicats
+        unique_content = list(set(self.content_urls))
+        
+        #gather group of valid urls
+        for filtred_urls in unique_content:
+            wideo = filtred_urls.startswith(f'{self.wideo_link}')
+            wiadomosc = filtred_urls.startswith(f'{self.wiadomosc_link}')
+            if wideo == True:
+                self.final_urls.append(filtred_urls)
+            elif wiadomosc == True:
+                self.final_urls.append(filtred_urls)
+                
+        #gather group of valid content of each url
+        for content_url in self.final_urls:
+            content_page = requests.get(content_url, headers=headers)
+            content_soup = BeautifulSoup(content_page.content, "html.parser")
+            content_results = content_soup.find("div", class_= f'{self.content_class}')   
+            self.particular_page_content.append(content_results.text)
+            
+        #return data as json
+        return {
+                "collected_data":{
+                    "Page": f"Wiadomości pochodzą z serwisu: {websiteLinks}",
+                    "Title": self.page_content,
+                    "Amount of articles": len(self.page_content),
+                    "Urls": self.final_urls,
+                    "Amount of urls": len(self.final_urls),
+                    "Content": self.particular_page_content,
+                }
+        }
+        
+    def tvn24_scrap(self):
+        websiteLinks = "TVN24"
+        
+        #elimination of invalid scraps 
+        for link in self.results.find_all('a'):
+            scrapido = link.find_all(class_ = f'{self.check_urls}')
+            if len(scrapido) != 0:
+                self.content_urls.append(link.get('href'))
+                
+        for content_url in self.content_urls:
+            content_page = requests.get(content_url,headers=headers)
+            content_soup = BeautifulSoup(content_page.content, "html.parser")
+            content_results = content_soup.find("div", class_=f'{self.content_class}')   
+            self.particular_page_content.append(content_results.text)
+                
+        return {
```

Comment:
Hey change your output to something like this:
```json{
  "MatchedNews": [
    {
      "Tvn": [
        {
          "Title": "value",
          "Content": "value",
          "Url": "value"
        }
      ],
      "Dziennik": [
        {
          "Title": "value",
          "Content": "value",
          "Url": "value"
        }
      ],
      "Polsat": [
        {
          "Title": "value",
          "Content": "value",
          "Url": "value"
        }
      ]
    }
  ]
}

Reply:
Resolved and updated